posts/2021-10-01-ga-release-post.adoc

```diff
@@ -0,0 +1,170 @@
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18559[Liberty is vulnerable to Information Disclosure (CVE-2021-29842 CVSS 3.7)]
++
+In the 21.0.0.9 release, an Information Disclosure vulnerability was discovered which could cause sensitive information to unintentionally be revealed. This issue has now been fixed in Open Liberty 21.0.0.10.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18475[Servlet ReadListener does not receive all HTTP request data]
++
+In 21.0.0.9 an issue was discovered where in rare cases, an app's `javax.servlet.ReadListener` could be invoked with `onAllDataRead()` prematurely, before all of the bytes sent along with the request have been made available to the app. In this scenario, `onDataAvailable()` would have been incorrectly invoked after the premature `onAllDataRead()` to provide notification for any remaining request data. This issue has been fixed by setting the `HttpServiceContextImpl` body to complete later on in the code.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18552[JAX-RS 2.0 and 2.1 implementation is executing resource method when Content-Type or Accept header contains invalid values]
++
+An issue was discovered in 21.0.0.9 which meant that JAX-RS (2.0 and 2.1) apps would execute resource code when bad requests were made using invalid `Content-Type` or `Accept` headers, rather than returning a 400 (Bad Request). A fix has now been implemented to ensure that if a request is made with invalid `Content-Type` or `Accept` headers, the application returns 400 (Bad Request).
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18348[ContainerRequestContext.getAcceptableLanguages() - fails with IllegalArgumentException when invalid locales are specified in the Accept-Language header]
++
+A bug was discovered which meant that `ContainerRequestContext.getAcceptableLanguages()` would throw an `IllegalArgumentException` when requests were made with invalid locales specified in the `Accept-Language` header. This `getAcceptableLanguages()` call should succeed and return the list from the headers as submitted. This issue has now been fixed to ensure that the `Accept-Language` header is handled consistently.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18437[JSF throws ClassNotFoundException for o.a.m.el.convert.ValueExpressionToValueBinding]
++
+Previously, A `JSF` application using `jsf-2.2` or `jsf-2.3` could encounter a `ClassNotFoundException` for the internal `MyFaces` class `ValueExpressionToValueBinding`, even if that class wasn't directly referenced by the app. This issue occured because the `org.apache.myfaces.el.convert` package was not made available (by Liberty's `jsf-2.2` and `jsf-2.3` bundles) to the thread context `classloader`. This visibility issue was fixed by updating those bundles.
```

Comment:
I'd like to add a bit more information (which I've only just now added to OL issue #18437:

Previously, A JSF application using [Application.createValueBinding(String)](https://docs.oracle.com/javaee/6/api/javax/faces/application/Application.html#createValueBinding(java.lang.String)) and `jsf-2.2` or `jsf-2.3` could encounter a `ClassNotFoundException` for the internal `MyFaces` class `ValueExpressionToValueBinding` ...